### ISSUE TITLE
Allow network calls to custom domain

### ISSUE THREAD

**Issue by jashsayani** [aws-amplify/amplify-js#7905]
I am using `amazon-cognito-identity-js` and the docs say that the Amazon domain should be whitelisted for use by the web app: **https://github.com/aws-amplify/amplify-js/tree/master/packages/amazon-cognito-identity-js#network-configuration**

However, I have a custom domain `auth.mydomain.com` that I want network calls to be made to. I don't see a way to provide this in the API. Is there a workaround? If not, support for making calls to custom domains seems important.

**Comment by stocaaro** [comment]
Are you trying to use a custom domain on the hosted UI or on the Cognito service API endpoint?

Wondering if [this cognito feature](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html) is related to your question.

**Comment by jashsayani** [comment]
Yes, amazon lets you add and use a custom domain. But the JS library has no way to set the domain, so it uses amazon's default domain.
